packages/react-aria-components/docs/FileTrigger.mdx

```diff
@@ -0,0 +1,208 @@
+  --background-color-pressed: var(--spectrum-global-color-gray-100);
+  --text-color: var(--spectrum-alias-text-color);
+  --text-color-disabled: var(--spectrum-alias-text-color-disabled);
+  --focus-ring-color: slateblue;
+
+  color: var(--text-color);
+  background: var(--background-color);
+  border: 1px solid var(--border-color);
+  border-radius: 4px;
+  appearance: none;
+  vertical-align: middle;
+  font-size: 1.2rem;
+  text-align: center;
+  margin: 0;
+  outline: none;
+  padding: 4px 12px;
+
+  &[data-pressed] {
+    box-shadow: inset 0 1px 2px rgb(0 0 0 / 0.1);
+    background: var(--background-color-pressed);
+    border-color: var(--border-color-pressed);
+  }
+
+  &[data-focus-visible] {
+    border-color: var(--focus-ring-color);
+    box-shadow: 0 0 0 1px var(--focus-ring-color);
+  }
+
+  &:disabled {
+    border-color: var(--border-color-disabled);
+    color: var(--text-color-disabled);
+  }
+}
+
+@media (forced-colors: active) {
+  .react-aria-Button {
+    forced-color-adjust: none;
+    --border-color: ButtonBorder;
+    --border-color-pressed: ButtonBorder;
+    --border-color-disabled: GrayText;
+    --background-color: ButtonFace;
+    --background-color-pressed: ButtonFace;
+    --text-color: ButtonText;
+    --text-color-disabled: GrayText;
+    --focus-ring-color: Highlight;
+  }
+}
+```
+
+</details>
+
+## Features
+
+Access to a computer’s file system can be created in HTML using an `<input>` with a file `type` attribute. However, file input's on their own are not very useful since they only allow users to select files from their devices. As a result, file inputs are commonly used inside a `<form>` alongside  a `<button>` to submit the form. `FileTrigger` simplifies this by setting up the input element's configuration with reference to a pressable component, such as a button, for a customizable file upload. 
+
+* **Customizable** – Works with any pressable component meaning that it is compatible with pressable React Aria Components and React Spectrum Components.
+
+## Anatomy
+
+A `FileTrigger` wraps around an input element and a pressable child, usually a button or link, that allows the user to upload files. The input element is not displayed and has no effect on the layout. 
+
+If a visual label is not provided on the pressable child, then an `aria-label` or `aria-labelledby` prop must be passed to identify the link to assistive technology.
+
+### Composed Components
+A `FileTrigger` can use the following components, which may also be used standalone or reused in other components.
+
+<ExampleCard
+  url="Button.html"
+  title="Button"
+  description="A button allows a user to perform an action with a mouse, touch, or keyboard.">
+  <Button />
+</ExampleCard>
+
+<ExampleCard
+  url="Link.html"
+  title="Link"
+  description="A link allows a user to navigate to another page.">
+  <Link />
+</ExampleCard>
+
+## Props
+
+<PropTable component={docs.exports.FileTrigger} links={docs.links} />
```

Comment:
Looks like the aria labeling props are on here but not used. I think we can remove these from the TypeScript definition since they should just be added to the `<Button>` instead.

We should probably also add the `name` prop here and pass it to the `<input>` element so that the selected file can be submitted as part of a form.